crates/flycheck/src/command.rs

```diff
@@ -119,14 +119,20 @@ impl<T> fmt::Debug for CommandHandle<T> {
 impl<T: ParseFromLine> CommandHandle<T> {
     pub(crate) fn spawn(mut command: Command, sender: Sender<T>) -> std::io::Result<Self> {
         command.stdout(Stdio::piped()).stderr(Stdio::piped()).stdin(Stdio::null());
-        let mut child = command.group_spawn().map(JodGroupChild)?;
 
         let program = command.get_program().into();
         let arguments = command.get_args().map(|arg| arg.into()).collect::<Vec<OsString>>();
         let current_dir = command.get_current_dir().map(|arg| arg.to_path_buf());
 
-        let stdout = child.0.inner().stdout.take().unwrap();
-        let stderr = child.0.inner().stderr.take().unwrap();
+        let mut child = StdCommandWrap::from(command);
+        #[cfg(unix)]
+        child.wrap(process_wrap::std::ProcessSession);
```

Comment:
Despite the name, command-group <5 used sessions; that was one reason to make the new process-wrap crate which supports both (and more). So using sessions here will essentially not change RA.

Reply:
Thanks! Do you have any advice on sessions vs. process groups? I'd expect groups to be the right pick here, but I don't really have experience with these things.

Reply:
I only have more confusion to add, unfortunately. From reading specifications and APIs, groups should be perfectly fine for most uses, including this one. However, when I changed command-group in version 5 to (correctly, I thought) use groups instead of sessions, I experienced a lot of feedback of usecases breaking (downstream from command-group, e.g. in watchexec), often for very obscure reasons, appearing as seemingly-unrelated nonsense symptoms.

So my recommendation is that given sessions (i.e. command-group <5) are known to work, keep using that. Conceptually, it more closely approximates how the program would be run as from a shell.